src/org/usfirst/frc/team467/robot/TalonSpeedControllerGroup.java

```diff
@@ -0,0 +1,192 @@
+package org.usfirst.frc.team467.robot;
+
+import org.apache.log4j.Logger;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.SpeedController;
+
+public class TalonSpeedControllerGroup implements SpeedController {
+	private static final Logger LOGGER = Logger.getLogger(TalonSpeedControllerGroup.class);
+	WPI_TalonSRX leader;
+	WPI_TalonSRX follower1;
+	WPI_TalonSRX follower2;
+	
+	private int previousSensorPosition;
+	
+	ControlMode controlMode = ControlMode.PercentOutput;
+	
+	public TalonSpeedControllerGroup() {
+		RobotMap.HAS_WHEELS = false;
+		leader = null;
+		follower1 = null;
+		follower2 = null;
+	}
+	
+	public TalonSpeedControllerGroup(ControlMode _controlMode, boolean sensorIsInverted,
+			WPI_TalonSRX _leader, WPI_TalonSRX _follower1, WPI_TalonSRX _follower2) {
+		this.leader = _leader;
+		this.follower1 = _follower1;
+		this.follower2 = _follower2;	
+		this.controlMode = _controlMode;
+		
+		initMotor(leader);
+		initMotor(follower1);
+		initMotor(follower2);
+		
+		//only have sensor on leader
+		leader.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder, 0, RobotMap.TALON_TIMEOUT);
+		leader.setSensorPhase(sensorIsInverted);
+		leader.configMotionCruiseVelocity(1052 / 2, RobotMap.TALON_TIMEOUT); //1052 is 75 percent of the max speed, which is 1402	
+		leader.configMotionAcceleration(1052 / 2, RobotMap.TALON_TIMEOUT);		
+		zero();		
+	}
+	
+	private void initMotor(WPI_TalonSRX talon) {
+		talon.set(ControlMode.PercentOutput, 0);
+		talon.selectProfileSlot(0, 0);
+		talon.configAllowableClosedloopError(0, RobotMap.VELOCITY_ALLOWABLE_CLOSED_LOOP_ERROR, 0);
+
+		//Note: This was changed from voltage to percentage used with 1 representing 100 percent or max voltage 
+		//      and -1 representing 100 percent backwards.
+		talon.configNominalOutputReverse(0.0, 0);		
+		talon.configNominalOutputForward(0.0, 0);
+		talon.configPeakOutputForward(1.0, 0);
+		talon.configPeakOutputReverse(-1.0, 0);
+		
+		talon.configOpenloopRamp(0.2, RobotMap.TALON_TIMEOUT);
+		talon.configClosedloopRamp(0.2, RobotMap.TALON_TIMEOUT);		
+	}
+	
+	public void logClosedLoopErrors(String side) {
+		LOGGER.debug(
+				//TODO Check the arguments for the closed loop errors.
+				side + ": Vel = " + leader.getSelectedSensorVelocity(0) +
+				" Pos = " + leader.getSelectedSensorPosition(0) +
+				" Err = " + leader.getClosedLoopError(0));
+	}
+	
+	public void setPIDF(double p, double i, double d, double f){
+		if (!RobotMap.HAS_WHEELS) {
```

Comment:
I think a better check would be "if (leader != null)", that way we only depend on RobotMap.HAS_WHEELS once instead of over and over again. The relevant thing here is that you should not (can not) set values if leader is null.

The same comment applies to may methods below.